### PR TITLE
Rename sanity_checks to confidence_checks

### PR DESCRIPTION
**Pull request by AkshitaB**
Related: https://github.com/allenai/allennlp-models/pull/262

### CODE REVIEW

File: allennlp/sanity_checks/__init__.py [hunk 1]
```diff
@@ -1,2 +1,9 @@
-from allennlp.sanity_checks.verification_base import VerificationBase
-from allennlp.sanity_checks.normalization_bias_verification import NormalizationBiasVerification
+from allennlp.confidence_checks.verification_base import VerificationBase
+from allennlp.confidence_checks.normalization_bias_verification import NormalizationBiasVerification
+
+import warnings
+
+warnings.warn(
+    "Module 'sanity_checks' is deprecated, please use " "'confidence_checks' instead.",
```
Comment (epwalsh) [inline allennlp/sanity_checks/__init__.py:7]:
```suggestion
    "Module 'sanity_checks' is deprecated, please use 'confidence_checks' instead.",
```

File: allennlp/training/callbacks/confidence_checks.py [hunk 2]
```diff
@@ -2,25 +2,25 @@
...
+from allennlp.confidence_checks.normalization_bias_verification import NormalizationBiasVerification
 
 
 if TYPE_CHECKING:
     from allennlp.training.trainer import GradientDescentTrainer
 
 
-@TrainerCallback.register("sanity_checks")
-class SanityChecksCallback(TrainerCallback):
+@TrainerCallback.register("confidence_checks")
```
Comment (epwalsh) [inline allennlp/training/callbacks/confidence_checks.py:12]: Maybe should we should keep this also registered as "sanity_checks" for now?

```suggestion
@TrainerCallback.register("confidence_checks")
@TrainerCallback.register("sanity_checks")
```

File: allennlp/training/trainer.py [hunk 3]
```diff
@@ -304,6 +313,13 @@ def __init__(
             world_size=world_size,
         )
 
+        if run_sanity_checks is not None:
+            warnings.warn(
+                "'run_sanity_checks' is deprecated, please use " "'run_confidence_checks' instead.",
```
Comment (epwalsh) [inline allennlp/training/trainer.py:318]:
```suggestion
                "'run_sanity_checks' is deprecated, please use 'run_confidence_checks' instead.",
```

File: allennlp/training/trainer.py [hunk 4]
```diff
@@ -294,7 +302,8 @@ def __init__(
         num_gradient_accumulation_steps: int = 1,
         use_amp: bool = False,
         enable_default_callbacks: bool = True,
-        run_sanity_checks: bool = True,
+        run_sanity_checks: Optional[bool] = None,
```
Comment (epwalsh) [inline allennlp/training/trainer.py:305]: Maybe it's better to just check for this in `**kwargs`? Either way works, I just thought we had gone with the "kwargs" approach elsewhere. But I don't feel strongly either way.

Comment (AkshitaB) [reply allennlp/training/trainer.py]: Makes sense. I like the `**kwargs` approach. Updated, along with other suggestions.

**Review by epwalsh — APPROVED** [review]
LGTM!

**Review by ArjunSubramonian — APPROVED** [review]
This looks great! Thanks :)

**Pull request merged**